posthog/hogql/placeholders.py

```diff
@@ -35,14 +39,33 @@ def __init__(self, placeholders: Optional[dict[str, ast.Expr]]):
         self.placeholders = placeholders
 
     def visit_placeholder(self, node):
-        if not self.placeholders:
-            raise QueryError(f"Unresolved placeholder: {{{node.field}}}")
-        if node.field in self.placeholders and self.placeholders[node.field] is not None:
-            new_node = self.placeholders[node.field]
-            new_node.start = node.start
-            new_node.end = node.end
-            return new_node
+        # avoid circular imports
+        from common.hogvm.python.execute import execute_bytecode
+        from posthog.hogql.compiler.bytecode import create_bytecode
+
+        bytecode = create_bytecode(node.expr)
+        response = execute_bytecode(bytecode.bytecode, self.placeholders)
```

Comment:
Can't `visit_placeholder` visit non-hog expressions? What happens with the bytecode stuff in those cases? Or is any `ast.Expr` valid in bytecode?

Reply:
Any `ast.Expr` is valid bytecode, with the exception of some functions we can't call.